fastlane/Fastfile

```diff
@@ -854,36 +854,75 @@ desc 'Show current frameworks size'
 lane :show_frameworks_sizes do |options|
   next unless is_check_required(sources: sources_matrix[:size], force_check: @force_check)
 
-  options[:sizes] ||= frameworks_sizes
+  ['metrics/'].each { |dir| FileUtils.remove_dir(dir, force: true) }
+
+  sh("git clone #{metrics_git} #{File.dirname(sdk_size_path)}")
+  sdk_size_benchmarks = JSON.parse(File.read(sdk_size_path))
+  expected_sizes = sdk_size_benchmarks['develop']
+  actual_sizes = options[:sizes] || frameworks_sizes
+
+  markdown_table = "## SDK Size\n| `title` | `benchmark` | `branch` | `diff` | `status` |\n| - | - | - | - | - |\n"
+  (sdk_names << 'Total').each do |title|
+    is_total = title == 'Total'
+    benchmark_value = expected_sizes[title]
+    branch_value = actual_sizes[title.to_sym]
+    max_tolerance = 0.5 # Max Tolerance is 0.5MB
+    fine_tolerance = 0.25 # Fine Tolerance is 0.25MB
+
+    diff = branch_value - benchmark_value
+
+    status_emoji =
+      if is_total
+        ''
+      elsif diff < 0
+        outstanding_status
+      elsif diff > max_tolerance
+        fail_status
+      elsif diff > fine_tolerance
+        warning_status
+      else
+        success_status
+      end
 
-  if is_ci && !ENV['GITHUB_PR_NUM'].to_s.empty?
-    markdown_table = "## SDK Size\n| `StreamChat` | `StreamChatUI` | `Total` |\n| - | - | - |\n"
-    markdown_table << "|#{options[:sizes][:stream_chat]}MB|#{options[:sizes][:stream_chat_ui]}MB|#{options[:sizes][:total]}MB|"
-    sh("gh pr comment #{ENV.fetch('GITHUB_PR_NUM')} -b '#{markdown_table}'")
+    markdown_table << "|#{title}|#{benchmark_value}mb|#{branch_value}mb|#{diff}mb|#{status_emoji}|\n"
   end
 
-  FastlaneCore::PrintTable.print_values(
-    title: 'SDK Size',
-    config: {
-      StreamChat: "#{options[:sizes][:stream_chat]}MB",
-      StreamChatUI: "#{options[:sizes][:stream_chat]}MB",
-      'Total Size': "#{options[:sizes][:total]}MB"
-    }
-  )
+  FastlaneCore::PrintTable.print_values(title: 'Benchmark', config: expected_sizes)
+  FastlaneCore::PrintTable.print_values(title: 'SDK Size', config: actual_sizes)
+
+  if is_ci
+    pr_comment_required = !ENV['GITHUB_PR_NUM'].to_s.empty?
+    sdk_size_benchmarks[current_branch] = actual_sizes
+    File.write(sdk_size_path, JSON.pretty_generate(sdk_size_benchmarks))
+
+    Dir.chdir(File.dirname(sdk_size_path)) do
+      if sh('git status -s', log: false).to_s.empty?
+        pr_comment_required = false
+        UI.important('No changes in SDK sizes benchmarks. Skipping commit and comment.')
+      else
+        sh('git add -A')
+        sh("git commit -m 'Update #{sdk_size_path}: #{current_branch}'")
+        sh('git push')
+      end
```

Comment:
When is this run? 🤔

Reply:
Every time on CI:
- on PR
- nightly on develop